ciao-launcher/payload.go

```diff
@@ -168,11 +170,13 @@ func parseStartPayload(data []byte) (*vmConfig, *payloadError) {
 	vnicIP := strings.TrimSpace(net.PrivateIP)
 	sshPort := computeSSHPort(networkNode, vnicIP)
 	var volumes []volumeConfig
-	if start.Storage.ID != "" {
-		volumes = append(volumes, volumeConfig{
-			UUID:     start.Storage.ID,
-			Bootable: start.Storage.Bootable,
-		})
+	for _, storage := range start.Storage {
```

Comment:
We should return an error if storage.ID == "".  This check was needed in the past as we had a string and not an array so we needed to compare to the empty string to see whether a volume was specified.  But now we have a slice, the fact that there is an element in the slice means that someone has specified a volume.